pulsenotify/consumer.py

```diff
@@ -243,7 +243,10 @@ def log_data(self):
                 }
 
     def make_s3_key(self, run_id):
-        build_properties = self.definition['extra']['build_props']
+        # extra.build_props is for release promotion in releasetasks.
+        # payload.properties is for the in-tree one.
+        build_properties = self.definition['extra'].get('build_props') or \
+            self.definition['payload']['properties']
```

Comment:
We chatted and came to conclusion that we should upload logs only if either `extra.build_props` or `payload.properties` are set.